src/sysext/sysext.c

```diff
@@ -758,6 +723,29 @@ static int verb_merge(int argc, char **argv, void *userdata) {
         if (r < 0)
                 return log_error_errno(r, "Failed to discover extension images: %m");
 
+        HASHMAP_FOREACH(img, images) {
+                r = image_read_metadata(img);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read metadata for image %s: %m", img->name);
+        }
+
+        *ret_images = TAKE_PTR(images);
+
+        return 0;
+}
+
+static int verb_merge(int argc, char **argv, void *userdata) {
+        _cleanup_(hashmap_freep) Hashmap *images = NULL;
+        char **p;
+        int r;
+
+        if (!have_effective_cap(CAP_SYS_ADMIN))
+                return log_error_errno(SYNTHETIC_ERRNO(EPERM), "Need to be privileged.");
+
+        r = image_discover_and_read_metadata(&images);
+        if (r < 0)
+                return r;
```

Comment:
why read the metadata here btw?

Reply:
no particular reason apart that it's when they are discovered - would you prefer this to be somewhere else?

Reply:
but why would verb_merge() need the metdata? it didn't need it so far?

Reply:
It did need the metadata, but before this change the extension_release stuff was parsed live on the open tree after dissect_mount - but now it's discovered and stored as metadata on the Image/DissectedImage objects, just like the os_release stuff already is.

The reasons for this are that it seems to me those two sets of image metadata ought to be handled in the same way for consistency, and more importantly this will also be used with the ExtensionImages implementation where we want to figure the metadata story out before mounting the image in the final place - just like we do with os_release.